ml/src/main/scala/ai/h2o/sparkling/ml/params/HasPlugValues.scala

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ai.h2o.sparkling.ml.params
+
+import ai.h2o.sparkling.H2OContext
+import ai.h2o.sparkling.utils.SparkSessionUtils
+import org.apache.spark.sql.Row
+import org.apache.spark.sql.catalyst.expressions.GenericRow
+import org.apache.spark.sql.types._
+
+import scala.collection.JavaConverters._
+
+trait HasPlugValues extends H2OAlgoParamsBase {
+  private val plugValues = new DictionaryParam(
+    this,
+    "plugValues",
+    "A map containing values that will be used to impute missing values of the training/validation frame, " +
+      "use with conjunction missing_values_handling = PlugValues)")
+
+  setDefault(plugValues -> null)
+
+  def getPlugValues(): Map[String, Double] ={
```

Comment:
@wendycwong Can plug values contain anything else then `Double`? Can be plug values used for imputing values of categorical columns, or they could be applied only on numeric columns.

Reply:
@mn-mikke:  plug values just need to be the same as the column type.  If they are double, plug value should be double.  For enum, they will take integer values as plug values.  However, you can store the plug values to be double and just convert it to integer when you actually do the plugging in.